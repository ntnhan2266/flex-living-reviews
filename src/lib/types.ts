export interface HostawayReview {
  id: number;
  type: string;
  status: string;
  rating: number | null;
  publicReview: string;
  reviewCategory: {
    category: string;
    rating: number;
  }[];
  submittedAt: string;
  guestName: string;
  listingName: string;
}

export type Channel = 'hostaway' | 'airbnb' | 'booking' | 'google';
export type ReviewStatus =
  | 'awaiting'
  | 'pending'
  | 'scheduled'
  | 'submitted'
  | 'published'
  | 'expired'
  | 'approved'
  | 'public';

export interface NormalizedReview {
  id: string;
  propertyName: string;
  guestName: string;
  rating: number;
  comment: string;
  date: Date;
  channel: Channel;
  categories: Record<string, number | undefined>;
  status: ReviewStatus;
}

export interface PropertyStats {
  id: string;
  name: string;
  averageRating: number;
  totalReviews: number;
  approvedReviews: number;
  recentTrend: 'up' | 'down' | 'stable';
}

export type CategoryKey =
  | 'cleanliness'
  | 'communication'
  | 'location'
  | 'accuracy'
  | 'checkin'
  | 'value';

export type SortKey = 'date_desc' | 'date_asc' | 'rating_desc' | 'rating_asc';

export type Filters = {
  property: string;
  channel: '' | 'hostaway' | 'airbnb' | 'booking' | 'google';
  rating: '' | '1' | '2' | '3' | '4' | '5';
  startDate: string; // yyyy-mm-dd
  endDate: string;   // yyyy-mm-dd
  status: 'all' | 'pending' | 'approved';
  category?: '' | CategoryKey;
  minCategory?: '' | '1' | '2' | '3' | '4' | '5';
  sort?: SortKey;
};

export interface ApiResponse<T> {
  status: 'success' | 'error';
  data?: T;
  message?: string;
  total?: number;
}
