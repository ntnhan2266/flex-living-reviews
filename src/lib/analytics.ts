// src/lib/analytics.ts
import { NormalizedReview, ReviewStatus } from '@/lib/types';

export type Granularity = 'day' | 'week' | 'month';

export type TimeseriesDatum = {
  date: string; // "YYYY-MM-DD" for day/week start/month start
  count: number; // number of reviews in bucket
  avgRating: number; // 0..5 average in bucket (0 when no ratings)
};

/**
 * Buckets a JS Date into an ISO key by granularity:
 * - day  -> YYYY-MM-DD
 * - week -> YYYY-MM-DD of the week's Monday
 * - month-> YYYY-MM-01
 */
export function bucketKey(d: Date, granularity: Granularity): string {
  const dt = new Date(d);
  dt.setHours(0, 0, 0, 0);

  if (granularity === 'week') {
    // Monday as week start
    const day = dt.getDay() || 7; // 1..7 (Mon..Sun)
    if (day !== 1) dt.setDate(dt.getDate() - (day - 1));
  } else if (granularity === 'month') {
    dt.setDate(1);
  }

  // fastest way to get YYYY-MM-DD without deps
  return dt.toISOString().slice(0, 10);
}

/**
 * Build a timeseries for reviews.
 * @param reviews - list of reviews
 * @param options.statuses - optional statuses to include (e.g. ['approved'])
 * @param options.granularity - day|week|month (default 'day')
 */
export function getReviewTimeseries(
  reviews: NormalizedReview[],
  options?: { statuses?: ReviewStatus[]; granularity?: Granularity },
): TimeseriesDatum[] {
  const allowed = new Set(options?.statuses ?? []);
  const gran = options?.granularity ?? 'day';

  const grouped: Record<string, { count: number; sum: number; withRating: number }> = {};

  for (const r of reviews) {
    if (allowed.size && !allowed.has(r.status)) continue;

    // normalize date (handle string | Date)
    const d = r.date instanceof Date ? r.date : new Date(r.date);
    const key = bucketKey(d, gran);
    if (!grouped[key]) grouped[key] = { count: 0, sum: 0, withRating: 0 };

    grouped[key].count += 1;

    const val = Number(r.rating);
    if (Number.isFinite(val)) {
      grouped[key].sum += val;
      grouped[key].withRating += 1;
    }
  }

  return Object.entries(grouped)
    .sort(([a], [b]) => (a < b ? -1 : a > b ? 1 : 0))
    .map(([date, { count, sum, withRating }]) => ({
      date,
      count,
      avgRating: withRating ? Number((sum / withRating).toFixed(2)) : 0,
    }));
}
