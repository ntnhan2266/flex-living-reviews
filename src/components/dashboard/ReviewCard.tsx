import { NormalizedReview } from '@/lib/types';
import { CheckCircle, Clock, Star } from 'lucide-react';
import { useState } from 'react';

type ReviewCardProps = {
  review: NormalizedReview;
  onApprove: () => void;
  onReset: () => void;
};

const StarRating = ({ rating }: { rating: number }) => (
  <div className="flex items-center gap-1">
    {[1, 2, 3, 4, 5].map((n) => (
      <Star
        key={n}
        className={`w-4 h-4 ${n <= Math.round(rating) ? 'text-yellow-400 fill-yellow-400' : 'text-gray-300'}`}
      />
    ))}
    <span className="text-sm text-gray-600 ml-1">
      {Number.isFinite(rating) ? rating.toFixed(1) : '-'}
    </span>
  </div>
);

export default function ReviewCard({ review, onApprove, onReset }: ReviewCardProps) {
  const [expanded, setExpanded] = useState(false);

  const getChannelColor = (channel: string) => {
    const colors = {
      hostaway: 'bg-blue-100 text-blue-800',
      airbnb: 'bg-red-100 text-red-800',
      booking: 'bg-purple-100 text-purple-800',
      google: 'bg-green-100 text-green-800',
    } as const;
    return (colors as Record<string, string>)[channel] ?? 'bg-gray-100 text-gray-800';
  };

  const dateStr = new Date(review.date).toLocaleDateString(undefined, {
    year: 'numeric',
    month: 'short',
    day: '2-digit',
  });

  return (
    <div className="bg-white p-6 rounded-lg shadow-sm border border-gray-200 hover:shadow-md transition-shadow">
      {/* Header */}
      <div className="flex items-start justify-between mb-4">
        <div className="flex-1">
          <h4 className="font-semibold text-gray-900 mb-1">{review.propertyName}</h4>
          <p className="text-sm text-gray-600">by {review.guestName}</p>
        </div>
        <div className="flex items-center gap-2">
          <span
            className={`capitalize px-2 py-1 text-xs rounded-full font-medium ${getChannelColor(review.channel)}`}
          >
            {review.channel}
          </span>
          <span className="text-sm text-gray-900">{dateStr}</span>
        </div>
      </div>

      {/* Rating */}
      <div className="mb-4">
        <StarRating rating={review.rating} />
      </div>

      {/* Comment */}
      {review.comment && (
        <>
          <p className={`text-gray-700 mb-2 ${expanded ? '' : 'line-clamp-3'}`}>{review.comment}</p>
          {review.comment.length > 160 && (
            <button
              onClick={() => setExpanded((s) => !s)}
              className="text-sm text-blue-600 hover:underline"
            >
              {expanded ? 'Show less' : 'Show more'}
            </button>
          )}
        </>
      )}

      {/* Category Ratings */}
      {review.categories && Object.values(review.categories).some(Boolean) && (
        <div className="grid grid-cols-2 md:grid-cols-3 gap-4 mb-4 text-sm">
          {Object.entries(review.categories).map(([category, rating]) =>
            rating ? (
              <div key={category} className="flex justify-between items-center">
                <span className="text-gray-900 capitalize">{category.replace(/_/g, ' ')}</span>
                <div className="flex items-center gap-1">
                  <Star className="w-3 h-3 text-yellow-400 fill-yellow-400" />
                  <span className="font-medium text-gray-700">{rating}</span>
                </div>
              </div>
            ) : null,
          )}
        </div>
      )}

      {/* Status and Actions */}
      <div className="flex items-center justify-between pt-4 border-t border-gray-200">
        <div className="flex items-center gap-3">
          {review.status === 'approved' && (
            <span className="flex items-center gap-1 text-green-600 text-sm font-medium">
              <CheckCircle className="w-4 h-4" />
              Approved
            </span>
          )}
          {review.status === 'pending' && (
            <span className="flex items-center gap-1 text-yellow-600 text-sm font-medium">
              <Clock className="w-4 h-4" />
              Pending
            </span>
          )}
        </div>

        <div className="flex gap-2">
          {review.status === 'pending' ? (
            <>
              <button
                onClick={onApprove}
                className="px-3 py-1 bg-green-600 text-white text-sm rounded-md hover:bg-green-700 transition-colors"
              >
                Approve
              </button>
            </>
          ) : (
            <button
              onClick={onReset}
              className="px-3 py-1 bg-gray-600 text-white text-sm rounded-md hover:bg-gray-700 transition-colors"
            >
              Revoke
            </button>
          )}
        </div>
      </div>
    </div>
  );
}
