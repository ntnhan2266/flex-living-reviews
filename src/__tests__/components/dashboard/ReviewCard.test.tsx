import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ReviewCard from '@/components/dashboard/ReviewCard';
import { NormalizedReview } from '@/lib/types';

function makeReview(overrides: Partial<NormalizedReview> = {}): NormalizedReview {
  return {
    id: 'r1',
    propertyName: 'Spacious 2 Bed',
    guestName: 'Jane Doe',
    rating: 4.2,
    comment: 'Lovely stay, clean and convenient.',
    date: new Date('2025-01-08T10:00:00Z'),
    channel: 'hostaway',
    categories: { cleanliness: 5, communication: 4, location: undefined },
    status: 'pending',
    ...overrides,
  };
}

describe('<ReviewCard />', () => {
  it('renders key information', () => {
    const review = makeReview();
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    // Title, guest, channel badge, date
    expect(screen.getByText('Spacious 2 Bed')).toBeInTheDocument();
    expect(screen.getByText(/by Jane Doe/i)).toBeInTheDocument();
    expect(screen.getByText('hostaway')).toBeInTheDocument();

    // Rating number shown to 1 decimal
    expect(screen.getByText('4.2')).toBeInTheDocument();

    // Pending badge
    expect(screen.getByText(/Pending/i)).toBeInTheDocument();
    // Approve button visible for pending
    expect(screen.getByRole('button', { name: /Approve/i })).toBeInTheDocument();
  });

  it('calls onApprove when Approve is clicked (pending status)', async () => {
    const user = userEvent.setup();
    const review = makeReview({ status: 'pending' });
    const onApprove = jest.fn();
    render(<ReviewCard review={review} onApprove={onApprove} onReset={jest.fn()} />);

    await user.click(screen.getByRole('button', { name: /Approve/i }));
    expect(onApprove).toHaveBeenCalledTimes(1);
  });

  it('shows Revoke when approved and calls onReset on click', async () => {
    const user = userEvent.setup();
    const review = makeReview({ status: 'approved' });
    const onReset = jest.fn();
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={onReset} />);

    // Approved badge
    expect(screen.getByText(/Approved/i)).toBeInTheDocument();
    // Revoke button visible
    const revoke = screen.getByRole('button', { name: /Revoke/i });
    expect(revoke).toBeInTheDocument();

    await user.click(revoke);
    expect(onReset).toHaveBeenCalledTimes(1);
  });

  it('renders category ratings only for defined entries', () => {
    const review = makeReview({
      categories: { cleanliness: 5, communication: 4, location: undefined, value: 0 },
    });
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    // cleanliness & communication visible
    expect(screen.getByText(/cleanliness/i)).toBeInTheDocument();
    expect(screen.getByText(/communication/i)).toBeInTheDocument();

    // location is undefined -> not rendered
    expect(screen.queryByText(/location/i)).not.toBeInTheDocument();

    // value = 0 is falsy, component filters out -> not rendered
    expect(screen.queryByText(/\bvalue\b/i)).not.toBeInTheDocument();
  });

  it('shows "Show more/Show less" toggle for long comments', async () => {
    const user = userEvent.setup();
    const longComment = 'A'.repeat(200);
    const review = makeReview({ comment: longComment });
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    const toggle = screen.getByRole('button', { name: /Show more/i });
    expect(toggle).toBeInTheDocument();

    await user.click(toggle);
    expect(screen.getByRole('button', { name: /Show less/i })).toBeInTheDocument();
  });

  it('renders the correct number of filled stars based on rounded rating', () => {
    // rating 3.6 rounds to 4
    const review = makeReview({ rating: 3.6 });
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    const ratingContainer = screen.getByText('3.6').closest('div')!;
    expect(ratingContainer).toContainElement(ratingContainer.querySelector('svg.text-yellow-400.fill-yellow-400'));
    expect(ratingContainer).toContainElement(ratingContainer.querySelector('svg.text-gray-300'));
  });

  it('shows a hyphen for non-finite rating', () => {
    const review = makeReview({ rating: NaN });
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    expect(screen.getByText('-')).toBeInTheDocument();
  });

  it('applies hostaway channel color classes on the badge', () => {
    const review = makeReview({ channel: 'hostaway' });
    render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);

    const badge = screen.getByText('hostaway');
    // The mapping uses: bg-blue-100 text-blue-800 for hostaway
    expect(badge).toHaveClass('bg-blue-100');
    expect(badge).toHaveClass('text-blue-800');
  });

  it('matches snapshot', () => {
    const review = makeReview();
    const { asFragment } = render(<ReviewCard review={review} onApprove={jest.fn()} onReset={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
