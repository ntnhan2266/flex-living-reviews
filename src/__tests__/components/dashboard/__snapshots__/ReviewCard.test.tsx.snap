// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ReviewCard /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-6 rounded-lg shadow-sm border border-gray-200 hover:shadow-md transition-shadow"
  >
    <div
      class="flex items-start justify-between mb-4"
    >
      <div
        class="flex-1"
      >
        <h4
          class="font-semibold text-gray-900 mb-1"
        >
          Spacious 2 Bed
        </h4>
        <p
          class="text-sm text-gray-600"
        >
          by Jane Doe
        </p>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span
          class="capitalize px-2 py-1 text-xs rounded-full font-medium bg-blue-100 text-blue-800"
        >
          hostaway
        </span>
        <span
          class="text-sm text-gray-900"
        >
          Jan 08, 2025
        </span>
      </div>
    </div>
    <div
      class="mb-4"
    >
      <div
        class="flex items-center gap-1"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-star w-4 h-4 text-yellow-400 fill-yellow-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="lucide lucide-star w-4 h-4 text-yellow-400 fill-yellow-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="lucide lucide-star w-4 h-4 text-yellow-400 fill-yellow-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="lucide lucide-star w-4 h-4 text-yellow-400 fill-yellow-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="lucide lucide-star w-4 h-4 text-gray-300"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <span
          class="text-sm text-gray-600 ml-1"
        >
          4.2
        </span>
      </div>
    </div>
    <p
      class="text-gray-700 mb-2 line-clamp-3"
    >
      Lovely stay, clean and convenient.
    </p>
    <div
      class="grid grid-cols-2 md:grid-cols-3 gap-4 mb-4 text-sm"
    >
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-gray-900 capitalize"
        >
          cleanliness
        </span>
        <div
          class="flex items-center gap-1"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-star w-3 h-3 text-yellow-400 fill-yellow-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <span
            class="font-medium text-gray-700"
          >
            5
          </span>
        </div>
      </div>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-gray-900 capitalize"
        >
          communication
        </span>
        <div
          class="flex items-center gap-1"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-star w-3 h-3 text-yellow-400 fill-yellow-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <span
            class="font-medium text-gray-700"
          >
            4
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-between pt-4 border-t border-gray-200"
    >
      <div
        class="flex items-center gap-3"
      >
        <span
          class="flex items-center gap-1 text-yellow-600 text-sm font-medium"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-clock w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6v6l4 2"
            />
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
          Pending
        </span>
      </div>
      <div
        class="flex gap-2"
      >
        <button
          class="px-3 py-1 bg-green-600 text-white text-sm rounded-md hover:bg-green-700 transition-colors"
        >
          Approve
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
